pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	if dk.ActiveGate().IsRoutingEnabled() {
+		multiCap := capability.NewMultiCapability(dk)
+		oneAgentDNSEntry := capability.BuildDNSEntryPointWithoutEnvVars(dk.Name, dk.Namespace, multiCap)
+		oneAgentNoProxyValues = append(oneAgentNoProxyValues, oneAgentDNSEntry)
+	}
+
+	return &startup.SecretConfig{
+		ApiUrl:              dk.Spec.APIURL,
+		ApiToken:            getAPIToken(tokens),
+		PaasToken:           getPaasToken(tokens),
+		TenantUUID:          dk.Status.OneAgent.ConnectionInfoStatus.TenantUUID,
+		Proxy:               proxy,
+		NoProxy:             dk.FeatureNoProxy(),
+		OneAgentNoProxy:     strings.Join(oneAgentNoProxyValues, ","),
+		NetworkZone:         dk.Spec.NetworkZone,
+		SkipCertCheck:       dk.Spec.SkipCertCheck,
+		HasHost:             dk.OneAgent().IsCloudNativeFullstackMode(),
+		MonitoringNodes:     hostMonitoringNodes,
+		HostGroup:           dk.OneAgent().GetHostGroup(),
+		InitialConnectRetry: dk.FeatureAgentInitialConnectRetry(),
+		EnforcementMode:     dk.FeatureEnforcementMode(),
+		ReadOnlyCSIDriver:   dk.FeatureReadOnlyCsiVolume(),
+		CSIMode:             dk.OneAgent().IsCSIAvailable(),
+	}, nil
+}
+
+func getPaasToken(tokens corev1.Secret) string {
+	if len(tokens.Data[dtclient.PaasToken]) != 0 {
+		return string(tokens.Data[dtclient.PaasToken])
+	}
+
+	return string(tokens.Data[dtclient.ApiToken])
+}
+
+func getAPIToken(tokens corev1.Secret) string {
+	return string(tokens.Data[dtclient.ApiToken])
+}
+
+// getHostMonitoringNodes creates a mapping between all the nodes and the tenantUID for the host-monitoring dynakube on that node.
+// Possible mappings:
+// - mapped: there is a host-monitoring agent on the node, and the dynakube has the tenantUID set => user processes will be grouped to the hosts  (["node.Name"] = "dynakube.tenantUID")
+// - not-mapped: there is NO host-monitoring agent on the node => user processes will show up as individual 'fake' hosts (["node.Name"] = "-")
+// - unknown: there SHOULD be a host-monitoring agent on the node, but dynakube has NO tenantUID set => user processes will restart until this is fixed (node.Name not present in the map)
+//
+// Checks all the dynakubes with host-monitoring against all the nodes (using the nodeSelector), creating the above mentioned mapping.
+func (g *BootstrapperInitGenerator) getHostMonitoringNodes(dk *dynakube.DynaKube) (map[string]string, error) {
+	tenantUUID := dk.Status.OneAgent.ConnectionInfoStatus.TenantUUID
+
+	imNodes := map[string]string{}
+	if !dk.OneAgent().IsCloudNativeFullstackMode() {
+		return imNodes, nil
+	}
+
+	if g.canWatchNodes {
+		var err error
+
+		imNodes, err = g.calculateImNodes(dk, tenantUUID)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		updateImNodes(dk, tenantUUID, imNodes)
+	}
+
+	return imNodes, nil
+}
+
+func (g *BootstrapperInitGenerator) calculateImNodes(dk *dynakube.DynaKube, tenantUUID string) (map[string]string, error) {
```

Comment:
not needed

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/8d9deb9183a43f3c0402f76ec0be8738de41bb50